### PR TITLE
add DEBUG_LIBSTDCXX option to GNU Make gcc for libstdc++ debug

### PR DESCRIPTION
**Pull request by zingale**
this catches bounds errors in the standard library containers

## Checklist

The proposed changes:
- [ ] fix a bug or incorrect behavior in AMReX
- [ ] add new capabilities to AMReX
- [ ] changes answers in the test suite to more than roundoff level
- [ ] are likely to significantly affect the results of downstream AMReX users
- [ ] include documentation in the code and/or rst files, if appropriate

### CODE REVIEW

**Comment by zingale** [comment]
see: https://gcc.gnu.org/onlinedocs/libstdc++/manual/debug_mode_using.html

**Comment by zingale** [comment]
not sure if we need this as an option so I left it WIP.  It doesn't find issues in Castro.  I am not sure how much of the standard library containers we rely on.

**Comment by WeiqunZhang** [comment]
Since the new option is opt-in, there is no harm I guess.

File: Tools/GNUMake/comps/gnu.mak [hunk 1]
```diff
@@ -103,6 +103,10 @@ else
   endif
 endif
 
+ifeq ($(DEBUG_LIBSTDCXX),TRUE)
+   CXXFLAGS += -D_GLIBCXX_DEBUG
```
Comment (WeiqunZhang) [inline Tools/GNUMake/comps/gnu.mak]:
```suggestion
   CPPFLAGS += -D_GLIBCXX_DEBUG
```

**Review by WeiqunZhang — APPROVED** [review]

**Pull request merged**